dpu-tools

```diff
@@ -0,0 +1,278 @@
+                    logger.error(f"Error ssh try: {result.err}")
+                    logger.error(f"Exception with minicom: {e}")
+                    logger.error("Exiting...")
+                    sys.exit(result.returncode)
+            fw = IPUFirmware(
+                self.args.imc_address,
+                version=result.out,
+                repo_url=self.args.repo_url,
+                dry_run=self.args.dry_run,
+                verbose=self.args.verbose,
+            )
+            fw.reflash_ipu()
+        elif self.dpu_type == DPUType.BF.name:
+            bf_fw = BFFirmware(self.args.bf_id)
+            bf_fw.firmware_reset()
+
+    def firmware_version(self) -> None:
+        if self.dpu_type == DPUType.IPU.name:
+            result = get_current_version(self.args.imc_address, logger=logger)
+            if result.returncode:
+                logger.debug("Failed with ssh, trying minicom!")
+                try:
+                    minicom_get_version(logger=logger)
+                except Exception as e:
+                    logger.error(f"Error ssh try: {result.err}")
+                    logger.error(f"Exception with minicom: {e}")
+                    logger.error("Exiting...")
+                    sys.exit(result.returncode)
+            print(result.out)
+        elif self.dpu_type == DPUType.BF.name:
+            bf_fw = BFFirmware(self.args.bf_id)
+            bf_fw.firmware_version()
+
+    def console(self) -> None:
+        logger.debug(f"Current DPU Type: {self.dpu_type}")
+        # TODO Since we can only console into the ipu through the provisioner, the dpu type should be given as an argument
+        if self.args.dpu_type is None:
+            logger.error("Console requires --dpu-type to function")
+            exit(1)
+        if self.dpu_type == DPUType.IPU.name:
+            console_ipu(self.args)
+        elif self.dpu_type == DPUType.BF.name:
+            console_bf(self.args)
+        elif self.dpu_type == DPUType.OCTEON.name:
+            logger.error("Marvel Octeon console support has not yet been implemented!")
+        else:
+            logger.error("UNKNOWN Type")
+
+    def list_dpus(self) -> None:
+        """
+        This function
+        """
+        devs = scan_for_dpus()
+        print("ID     netdev    PCI-Address   Kind")
+        print("-----  --------  ------------  ------")
+        for i, (k, (d, kind)) in enumerate(devs.items()):
+            print(f"{i: 5d}  {k.ljust(8)}  {d.ljust(12)}  {kind}")
+
+    def mode(self) -> None:
+        if self.args.set_mode:
+            bf_set_mode(self.args.bf_id, self.args.set_mode)
+        else:
+            bf_get_mode(self.args.bf_id, self.args.next_boot)
+
+    def pxeboot(self) -> None:
+        px = Pxeboot(self.args)
+        px.start_pxeboot()
+
+    def bfb(self) -> None:
+        if self.dpu_type == DPUType.BF.name:
+            download_bfb(self.args.bf_id)
+
+
+def main() -> None:
+    detected = detect_dpu_type()
```

Comment:
We should take the DPU type from the user if they provide it, otherwise we use the auto-detected dpu type.

We could even through a warning if the users dpu type and the autodetected do not match.